libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -161,14 +161,19 @@ void ImmutableValidator::endVisit(Return const& _return)
 bool ImmutableValidator::analyseCallable(CallableDeclaration const& _callableDeclaration)
 {
 	FunctionDefinition const* prevConstructor = m_currentConstructor;
+	ContractDefinition const* prevConstructorContract = m_currentConstructorContract;
```

Comment:
```suggestion
	ScopedSaveAndRestore constructorGuard{m_currentConstructorContract, nullptr};
```
and remove the manual restore. Same for the line above...